tests/functional/modules/test_zos_unarchive.py

```diff
@@ -0,0 +1,456 @@
+            record_format=record_format,
+        )
+        # Create members if needed
+        if data_set.get("dstype") in ["pds", "pdse"]:
+            for member in data_set.get("members"):
+                hosts.all.zos_data_set(
+                    name=f"{data_set.get('name')}({member})",
+                    type="member",
+                    state="present"
+                )
+        # Write some content into src
+        test_line = "this is a test line"
+        for member in data_set.get("members"):
+            if member == "":
+                ds_to_write = f"{data_set.get('name')}"
+            else:
+                ds_to_write = f"{data_set.get('name')}({member})"
+            hosts.all.shell(cmd=f"decho '{test_line}' \"{ds_to_write}\"")
+
+        format_dict = dict(name=format)
+        if format == "terse":
+            format_dict["format_options"] = dict(terse_pack="SPACK")
+        archive_result = hosts.all.zos_archive(
+            path=data_set.get("name"),
+            dest=MVS_DEST_ARCHIVE,
+            format=format_dict,
+        )
+        
+        # assert response is positive 
+        for result in archive_result.contacted.values():
+            print(result)
+            assert result.get("changed") is True
+            assert result.get("dest") == MVS_DEST_ARCHIVE
+            assert data_set.get("name") in result.get("archived")
+            cmd_result = hosts.all.shell(cmd = "dls {0}.*".format(HLQ))
+            for c_result in cmd_result.contacted.values():
+                assert MVS_DEST_ARCHIVE in c_result.get("stdout")
+        
+        hosts.all.zos_data_set(name=data_set.get("name"), state="absent")
+        
+        if format == "terse":
+            del format_dict["format_options"]["terse_pack"]
+        # Unarchive action
+        unarchive_result = hosts.all.zos_unarchive(
+            path=MVS_DEST_ARCHIVE,
+            format=format_dict,
+            remote_src=True,
+            dest_data_set=dict(name=data_set.get("name"),
+                               type=data_set.get("dstype"),
+                               record_format=record_format, 
+                               record_length=record_length)
+        )
+        
+        # assert response is positive 
+        for result in unarchive_result.contacted.values():
+            print(result)
+            assert result.get("changed") is True
+            assert result.get("failed", False) is False
+            # assert result.get("dest") == MVS_DEST_ARCHIVE
+            # assert data_set.get("name") in result.get("archived")
+            cmd_result = hosts.all.shell(cmd = "dls {0}.*".format(HLQ))
+            for c_result in cmd_result.contacted.values():
+                assert data_set.get("name") in c_result.get("stdout")
+        
+    finally:
+        hosts.all.zos_data_set(name=data_set.get("name"), state="absent")
+        hosts.all.zos_data_set(name=MVS_DEST_ARCHIVE, state="absent")
+
+
```

Comment:
Why is all this testing code commented out?

Reply:
Yeah there were still some test cases missing. They're in place now